store/tikv/mock-tikv/cluster.go

```diff
@@ -193,22 +205,140 @@ func (c *Cluster) Split(regionID, newRegionID uint64, key []byte, peerIDs []uint
 	c.Lock()
 	defer c.Unlock()
 
-	newRegion := c.regions[regionID].split(newRegionID, []byte(newMvccKey(key)), peerIDs, leaderPeerID)
+	newRegion := c.regions[regionID].split(newRegionID, []byte(NewMvccKey(key)), peerIDs, leaderPeerID)
 	c.regions[newRegionID] = newRegion
 }
 
-// Merge merges 2 Regions, their key ranges should be adjacent.
+// Merge merges 2 regions, their key ranges should be adjacent.
 func (c *Cluster) Merge(regionID1, regionID2 uint64) {
 	c.Lock()
 	defer c.Unlock()
 
-	c.regions[regionID1].merge(c.regions[regionID2].meta.GetEndKey())
+	c.regions[regionID1].merge(c.regions[regionID2].Meta.GetEndKey())
 	delete(c.regions, regionID2)
 }
 
+// SplitTable evenly splits the data in table into count regions.
+// Only works for single store.
+func (c *Cluster) SplitTable(mvccStore *MvccStore, tableID int64, count int) {
+	tableStart := tablecodec.GenTableRecordPrefix(tableID)
+	tableEnd := tableStart.PrefixNext()
+	c.splitRange(mvccStore, NewMvccKey(tableStart), NewMvccKey(tableEnd), count)
+}
+
+// SplitIndex evenly splits the data in index into count regions.
+// Only works for single store.
+func (c *Cluster) SplitIndex(mvccStore *MvccStore, tableID, indexID int64, count int) {
+	indexStart := tablecodec.EncodeTableIndexPrefix(tableID, indexID)
+	indexEnd := indexStart.PrefixNext()
+	c.splitRange(mvccStore, NewMvccKey(indexStart), NewMvccKey(indexEnd), count)
+}
+
+func (c *Cluster) splitRange(mvccStore *MvccStore, start, end MvccKey, count int) {
+	c.evacuateOldRegionRanges(start, end)
+	regionPairs := c.getEntriesGroupByRegions(mvccStore, start, end, count)
+	c.createNewRegions(regionPairs, start, end)
+}
+
+// getPairsGroupByRegions groups the key value pairs into splitted regions.
+func (c *Cluster) getEntriesGroupByRegions(mvccStore *MvccStore, start, end MvccKey, count int) [][]Pair {
+	startTS := uint64(math.MaxUint64)
+	limit := int(math.MaxInt32)
+	pairs := mvccStore.Scan(start.Raw(), end.Raw(), limit, startTS)
+	regionEntriesSlice := make([][]Pair, 0, count)
+	quotient := len(pairs) / count
+	remainder := len(pairs) % count
+	i := 0
+	for i < len(pairs) {
```

Comment:
Because `i` doesn't update in the same line, then init it there doesn't make much sense.